flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultPartitionManager.java

```diff
@@ -137,9 +211,46 @@ void onConsumedPartition(ResultPartition partition) {
 
 		// Release the partition if it was successfully removed
 		if (partition == previous) {
-			partition.release();
+			// move to cache if cachable
+			updateIntermediateResultPartitionCache(partition);
 
-			LOG.debug("Released {}.", partition);
+			LOG.debug("Cached {}.", partition);
```

Comment:
Let's adjust this log message to reflect that a partition might also be released after the update call.